src/cocoa_init.m

```diff
@@ -372,3 +371,43 @@ void _glfwPlatformTerminate(void)
         ;
 }
 
+static BOOL isNSApplicationRunning()
+{
+    BOOL isRunning = NO;
+    NSApplication *sharedNSApp = [NSApplication sharedApplication];
+    if(sharedNSApp)
+    {
+        isRunning = [sharedNSApp isRunning];
+    }
+    return isRunning;
+}
+
+void _glfwCreateAutoreleasePool()
+{
+    if(!isNSApplicationRunning())
+    {
+        _glfw.ns.autoreleasePool = [[NSAutoreleasePool alloc] init];
+    }
+}
+
+// Clears and resets the autorealease pool if GLFW is managing its own even loop,
+// and not using the built-in NSApplication event loop. GLFW creates its own
+// autorelease pool only if it is running its own event loop
```

Comment:
I think the second phrase is redundant, the first phrase contains already this info?